src/PyProxyHandler.cc

```diff
@@ -176,29 +176,806 @@ bool PyBaseProxyHandler::isExtensible(JSContext *cx, JS::HandleObject proxy,
+
+  JS::HandleValueArray jArgs(args);
+
+  JS::RootedValue jRet(cx);
+  if (!JS_CallFunctionName(cx, selfCopy, "lastIndexOf", jArgs, &jRet)) {
+    return false;
+  }
+
+  args.rval().setInt32(jRet.toInt32());
+  return true;
+}
+
+static bool array_splice(JSContext *cx, unsigned argc, JS::Value *vp) {
+  JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
+
+  JS::RootedObject proxy(cx, JS::ToObject(cx, args.thisv()));
+  if (!proxy) {
+    return false;
+  }
+  PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
+
+  uint64_t selfLength = (uint64_t)PyList_GET_SIZE(self);
+
+  int64_t relativeStart;
+  if (!JS::ToInt64(cx, args.get(0), &relativeStart)) {
+    return false;
+  }
+
+  /* actualStart is the index after which elements will be
+      deleted and/or new elements will be added */
+  uint64_t actualStart;
+  if (relativeStart < 0) {
+    actualStart = uint64_t(std::max(double(selfLength) + relativeStart, 0.0));
+  } else {
+    actualStart = uint64_t(std::min(double(relativeStart), double(selfLength)));
+  }
+
+  unsigned int argsLength = args.length();
+
+  /* insertCount is the number of elements being added */
+  uint32_t insertCount;
+  if (argsLength < 2) {
+    insertCount = 0;
+  }
+  else {
+    insertCount = argsLength - 2;
+  }
+
+  /* actualDeleteCount is the number of elements being deleted */
+  uint64_t actualDeleteCount;
+  if (argsLength < 1) {
+    actualDeleteCount = 0;
+  }
+  else if (argsLength < 2) {
+    actualDeleteCount = selfLength - actualStart;
+  }
+  else {
+    int64_t deleteCount;
+    if (!JS::ToInt64(cx, args.get(1), &deleteCount)) {
+      return false;
+    }
+
+    actualDeleteCount = uint64_t(std::min(std::max(0.0, double(deleteCount)), double(selfLength - actualStart)));
+  }
+
+  // get deleted items for return value
+  PyObject *deleted = PyList_GetSlice(self, actualStart, actualStart + actualDeleteCount);
+
+  // build list for SetSlice call
+  PyObject *inserted = PyList_New(insertCount);
+
+  JS::RootedObject *global = new JS::RootedObject(cx, JS::GetNonCCWObjectGlobal(proxy));
+  for (int index = 0; index < insertCount; index++) {
+    JS::RootedValue *elementVal = new JS::RootedValue(cx, args[index + 2].get());
+    PyList_SetItem(inserted, index, pyTypeFactory(cx, global, elementVal)->getPyObject());
+  }
+
+  PyList_SetSlice(self, actualStart, actualStart + actualDeleteCount, inserted);
+
+  args.rval().set(jsTypeFactory(cx, deleted));
+  return true;
+}
+
+static bool array_sort(JSContext *cx, unsigned argc, JS::Value *vp) {
+  JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
+
+  JS::RootedObject proxy(cx, JS::ToObject(cx, args.thisv()));
+  if (!proxy) {
+    return false;
+  }
+  PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
+
+  uint64_t selfLength = (uint64_t)PyList_GET_SIZE(self);
+
+  if (selfLength > 0) {
+    if (args.length() < 1) {
+      PyList_Sort(self);
```

Comment:
The behaviour with no arguments should be the same as `Array.prototype.sort()` with no arguments, where the elements first get converted to strings before being compared.
If the user wants the default python list.sort() behaviour with no arguments (which the elements are sorted using the `<` operator) they would have to call `list.sort()` through python.

Reply:
done